react/src/views/Boxes/BoxesView.test.tsx

```diff
@@ -0,0 +1,411 @@
+            id: "1",
+            name: "Lesvos",
+            organisation: {
+              __typename: "Organisation",
+              id: "1",
+              name: "BoxAid",
+            },
+          },
+          state: "Canceled",
+          targetBase: {
+            __typename: "Base",
+            id: "3",
+            name: "Samos",
+            organisation: {
+              __typename: "Organisation",
+              id: "2",
+              name: "BoxCare",
+            },
+          },
+        },
+      ],
+    },
+  },
+};
+
+const initialQueryNetworkError = {
+  request: {
+    query: BOXES_LOCATIONS_TAGS_SHIPMENTS_FOR_BASE_QUERY,
+    variables: {
+      baseId: "2",
+    },
+  },
+  result: {
+    errors: [new GraphQLError("Error!")],
+  },
+};
+
+describe("4.8.1 - Initial load of Page", () => {
+  it("4.8.1.1 - Is the Loading State Shown First?", async () => {
+    render(<Boxes />, {
+      routePath: "/bases/:baseId/boxes",
+      initialUrl: "/bases/2/boxes",
+      mocks: [initialQuery],
+      addTypename: true,
+      globalPreferences: {
+        dispatch: jest.fn(),
+        globalPreferences: {
+          organisation: { id: organisation2.id, name: organisation2.name },
+          availableBases: organisation1.bases,
+          selectedBase: { id: base2.id, name: base2.name },
+        },
+      },
+    });
+    // Test case 4.8.1.1
+    expect(screen.getByTestId("TableSkeleton")).toBeInTheDocument();
+  });
+
+  it("4.8.1.2 - Failed to Fetch Initial Data", async () => {
+    render(<Boxes />, {
+      routePath: "/bases/:baseId/boxes",
+      initialUrl: "/bases/2/boxes",
+      mocks: [initialQueryNetworkError],
+      addTypename: true,
+      globalPreferences: {
+        dispatch: jest.fn(),
+        globalPreferences: {
+          organisation: { id: organisation2.id, name: organisation2.name },
+          availableBases: organisation1.bases,
+          selectedBase: { id: base2.id, name: base2.name },
+        },
+      },
+    });
+    // Test case 4.8.1.2
+    expect(
+      await screen.findByText(/could not fetch boxes data! Please try reloading the page./i),
+    ).toBeInTheDocument();
+  });
+
+  it("4.8.1.3 - The Boxes Table is shown", async () => {
+    render(<Boxes />, {
+      routePath: "/bases/:baseId/boxes",
+      initialUrl: "/bases/2/boxes",
+      mocks: [initialQuery],
+      addTypename: true,
+      globalPreferences: {
+        dispatch: jest.fn(),
+        globalPreferences: {
+          organisation: { id: organisation2.id, name: organisation2.name },
+          availableBases: organisation1.bases,
+          selectedBase: { id: base2.id, name: base2.name },
+        },
+      },
+    });
+
+    // Test case 4.8.1.3
+    const row = await screen.findByRole("row", {
+      // eslint-disable-next-line max-len
+      name: /toggle all rows selected toggle sortby toggle sortby toggle sortby toggle sortby toggle sortby toggle sortby toggle sortby toggle sortby/i,
```

Comment:
lol @vahidbazzaz what's that?

Reply:
@HaGuesto  I will change it later, but I got this quick selector using the testing library chrome plugin :)